### PR TITLE
feat: [CI-13917]: Drone archive plugin

### PR DESCRIPTION
**Pull request by Anshika2203**
Execution link - https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/module/ci/orgs/default/projects/JenkinsPluginsImages/pipelines/Zip_Plugin/executions/BAasP278SAa9B66XfbfQWA/pipeline

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM golang:1.17
```
Comment (Ompragash) [inline docker/Dockerfile:1]: let's use lightweight alpine container with a tag eg: `alpine:3.19.4`

File: plugin/plugin.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type ZipPlugin struct {
+	Source   string `envconfig:"PLUGIN_SOURCE"`
+	Target   string `envconfig:"PLUGIN_TARGET"`
+	LogLevel string `envconfig:"PLUGIN_LOG_LEVEL"`
+}
```
Comment (Ompragash) [inline plugin/plugin.go:21]: let's also add `PLUGIN_FORMAT` as a setting and default it to zip as mentioned in the specs

File: plugin/plugin.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+func (p *ZipPlugin) Exec(ctx context.Context) error {
```
Comment (Ompragash) [inline plugin/plugin.go:23]: Also, let’s move the zip-specific code to a separate gzip folder, similar to the [cache plugin](https://github.com/drone-plugins/drone-meltwater-cache/tree/master/archive). We can follow the same approach for the new format types we’ll be adding next.

File: docker/Dockerfile.windows [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FROM golang:1.17
```
Comment (Ompragash) [inline docker/Dockerfile.windows:1]: you can use `plugins/base:windows-ltsc2022-amd64` for windows variant

File: docker/manifest.tmpl [hunk 5]
```diff
@@ -0,0 +1,25 @@
+image: harnesscommunitytest/archive-plugin:{{#if build.tag}}{{trimPrefix "v" build.tag}}{{else}}latest{{/if}}
```
Comment (Ompragash) [inline docker/manifest.tmpl:1]: remove harnesscommunity references and replace it with `plugins`

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,10 @@
+module example.com/m/v2
```
Comment (Ompragash) [inline go.mod:1]: please use `go mod init github.com/harness-community/drone-archive` to generate mod and sum files

**Review by Ompragash — COMMENTED** [review]
let's aim for a structure like this:
```
plugin
├── gzip
│   ├── main.go
│   └── main_test.go
├── tar
│   ├── main.go
│   └── main_test.go
└── zip
    ├── main.go
    └── main_test.go
```

File: docker/Dockerfile (hunk 1 above)
Comment (Anshika2203) [reply docker/Dockerfile]: Done

File: plugin/plugin.go (hunk 3 above)
Comment (Anshika2203) [reply plugin/plugin.go:21]: Done

File: go.mod (hunk 6 above)
Comment (Anshika2203) [reply go.mod]: Done

**Comment by Anshika2203** [comment]
> let's aim for a structure like this:
> 
> ```
> plugin
> ├── gzip
> │   ├── main.go
> │   └── main_test.go
> ├── tar
> │   ├── main.go
> │   └── main_test.go
> └── zip
>     ├── main.go
>     └── main_test.go
> ```

Added this structure.

File: docker/Dockerfile.windows (hunk 4 above)
Comment (Anshika2203) [reply docker/Dockerfile.windows]: Done

File: plugin/plugin.go (hunk 2 above)
Comment (Anshika2203) [reply plugin/plugin.go:19]: I will add PLUGIN_FORMAT as soon as I add multiple archive features - tar and gzip

File: docker/manifest.tmpl (hunk 5 above)
Comment (Anshika2203) [reply docker/manifest.tmpl:1]: I need access to plugins.

Comment (Anshika2203) [reply docker/manifest.tmpl]: Changed it to plugins/archive

**Review by Ompragash — APPROVED** [review]

**Pull request merged**